test/extended/images/sample_repos.go

```diff
@@ -0,0 +1,121 @@
+package images
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/test/e2e"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+// NewSampleRepoTest creates a function for a new ginkgo test case that will instantiate a template
+// from a url, kick off the buildconfig defined in that template, wait for the build/deploy,
+// and then confirm the application is serving an expected string value.
+func NewSampleRepoTest(repoName, templateURL, buildConfigName, serviceName, deploymentConfigName, expectedString, appPath string) func() {
```

Comment:
Maybe a `SampleConfig` struct with all these parameters? We can optionally add funcs to that struct if we need to perform more testing on various bits.